azure-kusto-ingest/test/e2eTest.js

```diff
@@ -0,0 +1,170 @@
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+
+        it('ingestFromStream', async function () {
+            for (let item of testItems) {
+                let stream = fs.createReadStream(item.path);
+                if (item.path.endsWith('gz')) {
+                    stream = new StreamDescriptor(stream, null, CompressionType.GZIP);
+                }
+                await new Promise((resolve) => {
+                    ingestClient.ingestFromStream(stream, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+    });
+
+    describe('StreamingIngestClient', function () {
+        it('ingestFromFile', async function () {
+            for (let item of testItems.filter(item => item.testOnstreamingIngestion)) {
+                await new Promise((resolve) => {
+                    streamingIngestClient.ingestFromFile(item.path, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+
+        it('ingestFromStream', async function () {
+            for (let item of testItems.filter(item => item.testOnstreamingIngestion)) {
+                let stream = fs.createReadStream(item.path);
+                if (item.path.endsWith('gz')) {
+                    stream = new StreamDescriptor(stream, null, CompressionType.GZIP);
+                }
+                await new Promise((resolve) => {
+                    streamingIngestClient.ingestFromStream(stream, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+    });
+});
+
+function sleep(ms) {
+    return new Promise((resolve) => { setTimeout(resolve, ms); });
+}
+
+function getTestResourcePath(name) {
+    return path.relative(process.cwd(), `azure-kusto-ingest/test/data/${name}`);
+}
+
+async function assertRowsCount(testItem) {
+    var count = 0;
+    var expected = testItem.rows;
+    for (var i = 0; i < 10; i++) {
+        await sleep(1000)
+        await new Promise((resolve) => {
+            queryClient.execute(databaseName, `${tableName} | count `, (err, results) => {
+                if (results) {
+                    count = results.primaryResults[0][0].Count - currentCount;
+                }
+                resolve()
+            });
+        });
+
+        if (count >= expected) {
```

Comment:
If count > expected, would it mean dup? Would we want to know about it?

Reply:
yes, or something wrong happened.. It will fail in the assertion, here is just to not waiting for timeout in this case